app/stats/[slug]/csv/route.ts

```diff
@@ -0,0 +1,39 @@
+import { usePapaParse } from "react-papaparse";
+
+import { getAllMetricSamples, type MetricSample } from "~/src/data/metrics";
+import { ContentType } from "~/src/utils";
+
+type Props = {
+  params: {
+    slug: string;
+  };
+};
+
+export async function GET(_: Request, { params }: Props): Promise<Response> {
+  const { slug } = params;
+
+  const { jsonToCSV } = usePapaParse();
+
+  const compareSamplesByDate = (a: MetricSample, b: MetricSample) =>
+    new Date(a.date).getTime() - new Date(b.date).getTime();
+
+  const allSamples = await getAllMetricSamples();
+  const samples = allSamples
+    .filter((s) => s.metricSlug === params.slug)
```

Comment:
Tady vlastně neřešíme 404 – když se uklepneš ve slugu, tak nedostaneš 404, ale prázdný dataset. Je mně to asi fuk, jen to pro jistotu zvědomuju.

Reply:
Právě, klidně bych to pustil z hlavy.